lpc17prl.lua

```diff
@@ -172,6 +174,32 @@ if opts.mode == 'read' then
 end
 
 
+if opts.mode == 'calculate' then
+  if not opts.fname then D.abort(2, "Binary file not provided") end
+
+  local f, err = io.open(opts.fname, "r+b")
+  if not f then D.abort(2, "Cannot open input file: "..opts.fname..": "..err) end
+
+  local sum = 0
+
+  for byteno=1,7,1 do
+    local bytes = f:read(4)
+    local int32 = B.dec32LE(bytes)
+    sum = sum + int32
+    D.yellow''(byteno,int32,B.hex(int32))
+  end
+
+  sum = -sum
+
+  f:seek("set", 28) -- ensure that we are writing checksum in correct place
+  f:write(B.enc32LE(sum))
+  f:close()
+
+  D.green'File saved! Checksum:'(sum,B.hex(sum))
+
+  os.exit()
```

Comment:
What if I'd like patch the file and write it to flash in one go?
Maybe we can think of supporting syntax like `./lpc17prl.lua -XW FILE`